src/main/java/edu/kis/powp/jobs2d/command/manager/parsing/JsonParser.java

```diff
@@ -0,0 +1,79 @@
+package edu.kis.powp.jobs2d.command.manager.parsing;
+
+import com.google.gson.Gson;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonParser implements Parser {
+
+	private static class JsonDriverCommand {
+		static class operation {
+			String opName;
+			int x;
+			int y;
+		}
+
+		List<operation> operations;
+	}
+
+
+	@Override
+	public DriverCommand parseFromString(String commandAsString) {
+		Gson gson = new Gson();
+
+		JsonDriverCommand jsonDriverCommand = gson.fromJson(commandAsString, JsonDriverCommand.class);
+		List<DriverCommand> driverCommands = new ArrayList<>();
+		for (JsonDriverCommand.operation op : jsonDriverCommand.operations)
+			if (op.opName.equals("SetPositionCommand")) driverCommands.add(new SetPositionCommand(op.x, op.y));
+			else if (op.opName.equals("OperateToCommand")) driverCommands.add(new OperateToCommand(op.x, op.y));
+
+		if (driverCommands.size() == 1)
+			return driverCommands.get(0);
+		return new CompoundCommand(driverCommands);
+	}
+
+	@Override
+	public String parseToString(DriverCommand driverCommand) {
+		Gson gson = new Gson();
+		JsonDriverCommand jsonDriverCommand = new JsonDriverCommand();
+
+		jsonDriverCommand.operations = getOperationFromDriverCommand(driverCommand);
+		return gson.toJson(jsonDriverCommand, JsonDriverCommand.class);
+	}
+
+	private List<JsonDriverCommand.operation> getOperationFromDriverCommand(DriverCommand driverCommand) {
+		List<JsonDriverCommand.operation> operationList = new ArrayList<>();
+		if (driverCommand instanceof OperateToCommand) {
```

Comment:
Only an optional suggestion: instanceof is not object-oriented approach. If you want to make it right you can use, e.g. visitor pattern.